pkg/services/ngalert/store/alert_rule.go

```diff
@@ -409,6 +410,7 @@ func (st DBstore) GetAlertRulesForScheduling(ctx context.Context, query *ngmodel
 			}
 			q = q.NotIn("org_id", excludeOrgs...)
 		}
+		q = q.Asc("namespace_uid", "rule_group", "rule_group_idx")
```

Comment:
Do you want to fall back to `id` sorting here if rule_group_idx is unset?

Reply:
It will not hurt